x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/index.ts

```diff
@@ -39,108 +28,67 @@ export function screenshotsObservableFactory(
 
   return function screenshotsObservable({
     logger,
-    url,
+    urls,
     conditionalHeaders,
     layout,
     browserTimezone,
-  }: ScreenshotObservableOpts): Rx.Observable<ScreenshotResults> {
-    const create$ = browserDriverFactory.create({
-      viewport: layout.getBrowserViewport(),
-      browserTimezone,
-    });
+  }: ScreenshotObservableOpts): Rx.Observable<ScreenshotResults[]> {
+    const create$ = browserDriverFactory.createPage(
+      { viewport: layout.getBrowserViewport(), browserTimezone },
+      logger
+    );
 
-    // @ts-ignore this needs to be refactored to use less random type declaration and instead rely on structures that work with inference TODO
-    return create$.pipe(
-      mergeMap(({ driver$, exit$ }) => {
-        const screenshot$ = driver$.pipe(
-          mergeMap(
-            (browser: HeadlessBrowser) => openUrl(browser, url, conditionalHeaders, logger),
-            browser => browser
-          ),
-          mergeMap(
-            (browser: HeadlessBrowser) => skipTelemetry(browser, logger),
-            browser => browser
-          ),
-          mergeMap(
-            (browser: HeadlessBrowser) => {
-              logger.debug(
-                'waiting for elements or items count attribute; or not found to interrupt'
-              );
+    return Rx.from(urls).pipe(
```

Comment:
Any reason to go with `concatMap` over `mergeMap`? I'm assuming it has to do with the utility return signature change (I did read over both in the rxjs docs)

Reply:
I've read over the docs quite a bit too, and I'll admit it is hard to understand how the difference will matter to us. The subscription pipeline handles one incoming event, and then is destroyed, so there are no incoming subsequent events that need to be in sync with the first one.

I think that is the reason I switched from mergeMap to concatMap - it seemed to make more sense with the fact that the pipeline is used for a single event, and then the subscription is cancelled. The next event gets a new subscription.

Thinking about it now, having the breakup of multiple `mergeMap` or `concatMap` is really about syntax and style. There could be just the `concatMap` on line 44 and it chains the async functions together and forms a payload, or cancels itself from the `exit$`.

Given how hard it was to try to justify, I think another change is due in this PR to simplify things here.

Reply:
I think there is some justifiable reasoning in keeping the pipeline separated using observable operators, instead of having a larger function with multiple async steps. The observable pipeline is cancelable, and we're racing the pipeline against the `exit$` observable.